app/workers/subject_removal_worker.rb

```diff
@@ -0,0 +1,12 @@
+require 'subjects/remover'
+
+class SubjectRemovalWorker
+  include Sidekiq::Worker
+
+  sidekiq_options queue: :data_medium
+
+  def perform(subject_id)
+    Subjects::Remover.new(subject_id).cleanup
+  rescue Subjects::Remover::NonOrphan
```

Comment:
Right - i was using it like AR.find that raises if it can't find the subject id (which we use a lot in the workers). I can rework it but from a client of the remover i thought this approach was reasonable.